scripts/purchaseServers.js

```diff
@@ -44,21 +44,26 @@ async function buyServers(ns) {
             serverCost = ns.getPurchasedServerCost(ram);
         }
         shopServer = await ns.prompt(`Would you like to buy a ${ram}GB server for ${ns.nFormat(serverCost, "$0.00a")}`);
-        if (shopServer) { shopServer = buyServer(ns, ram); }
+        if (shopServer) { shopServer = buyServer(ns, ram, myMoney, serverCost); }
```

Comment:
Since we know how much money the user has at this point, why don't we move the check up here? The we don't need to add the additional parameters to buyServer. We can avoid even asking them if they want to buy a server they can't.